### PR TITLE
Feature/api 

### PR DESCRIPTION
**Pull request by heeler**
What was addressed on this branch:

- docstring format changed to numpy
- changed int to size_t in pb_helper.h that was causing issues on very large files
- updated the API so that Y & X are returned re methods: dims, dims_shape, size
- updated the SubblockSortable to be aware of B index (incidental)
- added scene bounding box access method and scene height/width function (scenes can technically have different sizes but the users I have spoken to say this is hard to configure so maybe it's overkill)
- improved the exception messaging particularly in the case of overspecified dimensions.
- made running the C++ tests as part of the CMake build dependent on env variable RUN_TESTS if the variable isn't set the tests aren't auto-run (IDE convenience)

@JacksonMaxfield if you could comment on the CziFile.py and test_czi_file.py that would be great
@toloudis & others any feedback if you have time would be great. Thanks

### CODE REVIEW

File: pylibczi/CziFile.py [hunk 1]
```diff
@@ -134,9 +199,14 @@ def convert_to_buffer(file: types.FileLike) -> Union[BinaryIO, np.ndarray]:
             )
 
     def read_meta(self):
```
Comment (JacksonMaxfield) [inline pylibczi/CziFile.py]: I would change this to a property that reads once then stores like it does in AICSImage

File: pylibczi/CziFile.py [hunk 2]
```diff
@@ -186,10 +268,13 @@ def read_image(self, m_index: int = -1, **kwargs):
                        H = 7   # The H-dimension ("phase").
                        V = 8   # The V-dimension ("view").
 
-        :returns: a tuple of (numpy.ndarray, a list of (Dimension, size)) the second element of the tuple is to make
+        Returns
+        -------
+        (numpy.ndarray, [Dimension, Size])
+            a tuple of (numpy.ndarray, a list of (Dimension, size)) the second element of the tuple is to make
             sure the numpy.ndarray is interpretable. An example of the list is ::
```
Comment (JacksonMaxfield) [inline pylibczi/CziFile.py]: I think you forgot to remove the `::` at the end of this when switching to numpy format.

File: pylibczi/tests/test_czi_file.py [hunk 3]
```diff
@@ -60,7 +100,7 @@ def test_is_mosaic(data_dir, fname, expected):
 ])
 def test_destructor(data_dir, fname, expected):
     czi = CziFile(str(data_dir / fname))
-    czi = CziFile(str(data_dir / fname))
+    czi = CziFile(str(data_dir / fname))  # NOQA F841
```
Comment (JacksonMaxfield) [inline pylibczi/tests/test_czi_file.py]: I am assuming you got frustrated by flake8 and yelled "STOP QAing THIS" 😂

File: _pylibczi/Reader.cpp [hunk 4]
```diff
@@ -47,16 +65,76 @@ namespace pylibczi {
   /// @brief get_shape_from_fp returns the Dimensions of a ZISRAW/CZI when provided a ICZIReader object
   /// @param czi: a shared_ptr to an initialized CziReader object
   /// @return A Python Dictionary as a PyObject*
-  Reader::MapDiP
+  Reader::MapCDiP
```
Comment (toloudis) [inline _pylibczi/Reader.cpp]: what does MapCDiP mean?

File: pylibczi/CziFile.py [hunk 5]
```diff
@@ -186,21 +272,32 @@ def read_image(self, m_index: int = -1, **kwargs):
                        H = 7   # The H-dimension ("phase").
                        V = 8   # The V-dimension ("view").
 
-        :returns: a tuple of (numpy.ndarray, a list of (Dimension, size)) the second element of the tuple is to make
+                       flatten = True
```
Comment (toloudis) [inline pylibczi/CziFile.py]: suggest renaming `flatten` to `split_bgr`

**Pull request merged**